api_call_options.go

```diff
@@ -94,26 +98,42 @@ func (g *GetOptions) GetPage() int { // guaranteed int
 	return g.Page
 }
 
+func (g *GetOptions) GetCopy() *GetOptions {
```

Comment:
What do you think about `GetOptions() GetOptions`?  I can't help but think of https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#GetOptions and how the comparison differs because in those cases a DeepCopy does does not guarantee a GetOptions if the caller is nil.

The guarantee function we are making here is more like a `*string` to `string` getter:

```go
func (c *caller) GetName() String {
  if c == nil { return "" }
  if c.name == nil { return "" }
  return *c.name  
}
```

Except... We are both taking and returning a pointer. 

I was originally thinking `CopyOrNew` would be a more explicit name, but I can get behind `GetCopy`.
We should definitely include a comment describing this function as guaranteeing a  GetOptions response, which is either a copy of the caller or a new GetOptions if the caller is nil.

Reply:
Thinking about it, from `GetCopy`, `GetOptions` and `CopyOrNew`, I would choose `CopyOrNew` as that's saying exactly what it does.

Reply:
.. I wrote GetCopy because I needed it in func nextPage for forming the next-page URL with other opts (not straightforward because the meta.Next href will contain sth like `/projects?page=2`, no matter what opts you had in the original HTTP GET querystring). Considering we have to take the user-given opts from own variable, and we need to use the next page number, the next-page URL is not a great start for elegant URL building.

^ Maybe it could be suggested to the API devs: "Please include original querystring in the next-page href"

Reply:
I created a ticket for that - common parameters (such as sort_by, sort_dir, includes, excludes, and search) should be reproduced in pagination links.